File: 0382-linked-list-random-node/0382-linked-list-random-node.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    
    
    def __init__(self, head: Optional[ListNode]):
        self.head=head

    def getRandom(self) -> int:
        x=1
        ans=0
        curr=self.head
        while(curr):
            if(random.random()< 1/x):
                ans=curr.val
            
            x+=1
            curr=curr.next
        return ans


# Your Solution object will be instantiated and called as such:
# obj = Solution(head)
# param_1 = obj.getRandom()

File: 0382-linked-list-random-node/0382-linked-list-random-node.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode h;
    
    public Solution(ListNode head) {
        this.h=head;
    }
    
    public int getRandom() {
        int x=1;
        int r=0;
        ListNode c=this.h;
        while(c!=null){
            if(Math.random()<(1.0/x)){
                r=c.val;
            }
            x++;
            c=c.next;
        }
        return r;
    }
}

/**
 * Your Solution object will be instantiated and called as such:
 * Solution obj = new Solution(head);
 * int param_1 = obj.getRandom();
 */